### PR TITLE
fixed cannot load the correct url of nmrium wrapper

### PR DESCRIPTION
**Pull request by baolanlequang**
- [x] rather 1-story 1-commit than sub-atomic commits

- [x] commit title is meaningful =>  git history search

- [x] commit description is helpful => helps the reviewer to understand the changes

- [x] code is up-to-date with the latest developments of the target branch (rebased to it or whatever) => :fast_forward:-merge for linear history is favoured

- [x] added code is linted

- [x] tests are passing (at least locally): we still have some random test failure on CI. thinking of asking spec/examples.txt to be commited

- [ ] in case the changes are visible to the end-user,  video or screenshots should be added to the PR => helps  with user testing

- [ ] testing coverage improvement is improved.

- [ ] CHANGELOG :  add a bullet point on top (optional: reference to github issue/PR )

- [ ] parallele PR for documentation  on docusaurus  if the feature/fix is tagged for a release

### CODE REVIEW

File: app/api/chemotion/chem_spectra_api.rb [hunk 1]
```diff
@@ -208,7 +208,8 @@ def raw_file(att)
               protocol = nmrium_uri.scheme
               host = nmrium_uri.host
               port = nmrium_uri.port == 443 ? '' : nmrium_uri.port
-              { protocol: protocol, host: host, port: port }
+              nmrium_origin = port == '' ? "#{protocol}://#{host}" : "#{protocol}://#{host}:#{port}"
```
[the remark this answers is not in the document]
Comment (baolanlequang) [reply app/api/chemotion/chem_spectra_api.rb:211]: I want to do it in the backend, the frontend only needs to get the values and compare. The `nmrium_origin` is used in the frontend to fill the message from nmrium wrapper.
The comparison of port value is to make sure we don't use port 443 in the past.
But as you suggested, we're using `nmriumwrapper` in the `spectra.yml` instead of separated values, I leave it as to cover the older config file.
I am open to changing this, but is it necessary or not?

File: app/packs/src/components/nmriumWrapper/NMRiumDisplayer.js [hunk 2]
```diff
@@ -95,15 +92,27 @@ export default class NMRiumDisplayer extends React.Component {
...
+          let nmriumData = null;
+          if (eventData.data) {
+            const { state } = eventData.data;
+            if (state) {
+              nmriumData = state;
+            }
+            else {
+              nmriumData = eventData.data;
+            }
+          }
```
Comment (ptrxyz) [inline app/packs/src/components/nmriumWrapper/NMRiumDisplayer.js:105]: Hm, the whole `if` tree looks a complex to me, making it hard to follow:
image

can we simplify the code to:

```
nmriumData = eventData.data?.state || eventData.data;
```

image

This leads to the same results except that in the first two cases, we end up with undefined instead of null, which could probably further been worked around with `nmriumData = ( ....the line above... ) || null` in case it makes a difference later in the code.

Comment (baolanlequang) [reply app/packs/src/components/nmriumWrapper/NMRiumDisplayer.js:105]: Yes, we can keep it simple. But I doubt is it clear to follow up, especially when we will update our site based on nmrium-wrapper returned data.

File: app/packs/src/components/nmriumWrapper/NMRiumDisplayer.js [hunk 3]
```diff
@@ -95,15 +92,27 @@ export default class NMRiumDisplayer extends React.Component {
...
+          
+          if (!nmriumData) {
```
[the remark this answers is not in the document]
Comment (ptrxyz) [reply app/packs/src/components/nmriumWrapper/NMRiumDisplayer.js:110]: This one is still pending. Can you please add some explaining comments or maybe even proper JSDoc annotations to make the types somehow understandable to someone from outside?

File: app/api/chemotion/chem_spectra_api.rb [hunk 4]
```diff
@@ -201,14 +201,10 @@ def raw_file(att)
...
+              { nmrium_url: '' }
             else
               nmrium_url = Rails.configuration.spectra.nmriumwrapper.url
-              nmrium_uri = URI(nmrium_url)
-              protocol = nmrium_uri.scheme
-              host = nmrium_uri.host
-              port = nmrium_uri.port == 443 ? '' : nmrium_uri.port
-              { protocol: protocol, host: host, port: port }
+              { nmrium_url: nmrium_url }
             end
```
Comment (ptrxyz) [inline app/api/chemotion/chem_spectra_api.rb:208]: Starting from 203 following, couldn't this simply be:

```rb
{ nmrium_url: Rails.configuration.spectra.nmriumwrapper.url.presence || '' }
```

Maybe even without the `|| ''`?

**Review by ptrxyz — CHANGES_REQUESTED** [review]

File: app/api/chemotion/chem_spectra_api.rb (hunk 4 above)
Comment (baolanlequang) [reply app/api/chemotion/chem_spectra_api.rb]: @ptrxyz I updated

**Review by PiTrem — APPROVED** [review]
